File: frontend/src/models/Auction.ts
export type AuctionPost = {
  startDate: Date;
  endDate: Date;
  artId: number;
};

export type AuctionGet = {
  id: number;
  startDate: Date;
  endDate: Date;
  artId: number;
  status: string;
  createdDate: Date;
};

File: backend/Models/Auction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    [Table("Auction")]
    public class Auction
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public string Status { get; set; } = string.Empty;
        [Required]
        public DateTime StartDate { get; set; }
        [Required]
        public DateTime EndDate { get; set; }
        [Required]
        [ForeignKey("Art")]
        public int ArtId { get; set; }
        public Art Art { get; set; }
        [Required]
        [ForeignKey("ApplicationUser")]
        public string UserId { get; set; } = string.Empty;
        public User User { get; set; }
        public List<Bid> Bids { get; set; }
        public DateTime CreatedDate { get; set; } = DateTime.Now;

    }
}
